devdoc/thandle_requirements.md

```diff
@@ -165,22 +165,54 @@ Given a previously constructed `THANDLE(T)`, `THANDLE_GET_T(T)` reeturns a point
 
 **SRS_THANDLE_02_024: [** `THANDLE_GET_T(T)` shall return the same pointer as `THANDLE_MALLOC`/`THANDLE_MALLOC_WITH_EXTRA_SIZE` returned at the handle creation time. **]**
 
-###     THANDLE_CREATE_FROM_CONTENT_MACRO(T)
+### THANDLE_CREATE_FROM_CONTENT_FLEX_MACRO(T)
+```c
+THANDLE_CREATE_FROM_CONTENT_FLEX_MACRO(T)
+THANDLE(T) THANDLE_CREATE_FROM_CONTENT_FLEX(T)(const T* source, void(*dispose)(T*), int(*copy)(T* destination, const T* source), size_t(*get_sizeof)(const T* source))
+```
+
+Given a previously existing T, `THANDLE_CREATE_FROM_CONTENT_FLEX` will copy `T`'s content and return a `THANDLE(T)`. The allocated memory's size for copy is given by `get_sizeof` return value.
+
+**SRS_THANDLE_02_025: [** If `source` is `NULL` then `THANDLE_CREATE_FROM_CONTENT_FLEX` shall fail and return `NULL`. **]**
+
+**SRS_THANDLE_02_031: [** `THANDLE_CREATE_FROM_CONTENT_FLEX` shall call `get_sizeof` to get the needed size to store `T`. **]**
+
+**SRS_THANDLE_02_026: [** `THANDLE_CREATE_FROM_CONTENT_FLEX` shall allocate memory. **]**
+
+**SRS_THANDLE_02_027: [** If `copy` is `NULL` then `THANDLE_CREATE_FROM_CONTENT_FLEX` shall memcpy the content of `source` in allocated memory. **]**
+
+**SRS_THANDLE_02_028: [** If `copy` is not `NULL` then `THANDLE_CREATE_FROM_CONTENT_FLEX` shall call `copy` to copy `source` into allocated memory. **]**
+
+**SRS_THANDLE_02_029: [** `THANDLE_CREATE_FROM_CONTENT_FLEX` shall initialize the ref count to 1, succeed and return a non-`NULL` value. **]**
+
+**SRS_THANDLE_02_030: [** If there are any failures then `THANDLE_CREATE_FROM_CONTENT_FLEX` shall fail and return `NULL`. **]**
+
+###  THANDLE_CREATE_FROM_CONTENT_MACRO(T)
 ```c
 THANDLE_CREATE_FROM_CONTENT_MACRO(T)
 THANDLE(T) THANDLE_CREATE_FROM_CONTENT(T)(const T* source, void(*dispose)(T*), int(*copy)(T* destination, const T* source))
 ```
 
-Given a previously existing T, `THANDLE_CREATE_FROM_CONTENT` will copy `T`'s content and return a `THANDLE(T)`.
+`THANDLE_CREATE_FROM_CONTENT_MACRO` is the same as `THANDLE_CREATE_FROM_CONTENT_FLEX_MACRO` where `get_sizeof` function returns what the C's sizeof operator evaluates to.
+
+**SRS_THANDLE_02_032: [** `THANDLE_CREATE_FROM_CONTENT` returns what `THANDLE_CREATE_FROM_CONTENT_FLEX(T)(source, dispose, copy, THANDLE_GET_SIZEOF(T));` returns. **]**
```

Comment:
>THANDLE_GET_SIZEOF [](start = 129, length = 18)

I don't see this anywhere else in the doc (but I can assume it's really `sizeof` based on the description above).

Reply:
added an explicit chapter for THANDLE_GET_SIZEOF

---
In reply to: [411487113](https://github.com/Azure/azure-c-util/pull/28#discussion_r411487113) [](ancestors = 411487113)